src/power_grid_model/core/data_handling.py

```diff
@@ -8,7 +8,7 @@
 
 
 from enum import Enum
-from typing import Mapping, Tuple, Union
+from typing import Mapping, Tuple
```

Comment:
please double-check that all cases of List, Tuple, Set, Dict have been removed
```suggestion
from typing import Mapping
```